### PR TITLE
ENH: make benchopt run . -s solver[option=12]   work

### LINKED ISSUE #178: ENH : refine solver name matching
currently `-s celer` also selects `python-pgd[use_acceleration=True]` because it contains `celer`

### PR DESCRIPTION
**Pull request by mathurinm**
the many calls to replace are unsatisfying  but short, readable and not in a part of the code which is speed critical

closes #178

### CODE REVIEW

**Review by agramfort — APPROVED** [review]
@tomMoral i let you the green button :)

**Comment by tomMoral** [comment]
Maybe it would be nice to take the opportunity to update the doc on using patterns to select solver/dataset?
If you have the time @mathurinm I will delay the merge of the PR, if not, let's merge and see later.

File: doc/index.rst [hunk 1]
```diff
@@ -61,29 +61,45 @@ This benchmark can then be retrieved locally with:
...
     $ benchopt install --env ./benchmark_lasso
 
-To run Lasso benchmarks on all datasets and with all solvers, run:
+To run benchmarks on all datasets and with all solvers, run:
 
 .. code-block::
 
     $ benchopt run --env ./benchmark_lasso
 
+To run only the `sklearn` and `celer` solvers, on the `simulated` and `finance` datasets, run:
```
Comment (tomMoral) [inline doc/index.rst:81]: typo rst <-> markdown
```suggestion
To run only the ``sklearn`` and ``celer`` solvers, on the ``simulated`` and ``finance`` datasets, run:
```

File: doc/index.rst [hunk 2]
```diff
@@ -61,29 +61,45 @@ This benchmark can then be retrieved locally with:
...
+.. code-block::
+
+    $ benchopt run --env ./benchmark_lasso -s sklearn -s celer -d simulated -d finance
+
+Some solvers and dataset have parameters; by default all combinations are run.
+If you want to run a specific configuration, pass it explicitly, e.g., to run the python-pgd solver only with its parameter ``use_acceleration`` set to True, use:
+
+.. code-block::
+
+    $ benchopt run --env ./benchmark_lasso -s python-pgd[use_acceleration=False]
```
Comment (tomMoral) [inline doc/index.rst:92]: typo
```suggestion
    $ benchopt run --env ./benchmark_lasso -s python-pgd[use_acceleration=True]
```

File: doc/index.rst [hunk 3]
```diff
@@ -159,3 +175,4 @@ Contents
 .. _`lasso`: https://github.com/benchopt/benchmark_lasso
 .. _`logreg_l1`: https://github.com/benchopt/benchmark_logreg_l1
 .. _`logreg_l2`: https://github.com/benchopt/benchmark_logreg_l2
+.. _`cli_documentation`: https://benchopt.github.io/cli.html
```
Comment (tomMoral) [inline doc/index.rst:178]: Not necessary, this is already defined in the `cli.rst` file I think.

**Review by tomMoral — APPROVED** [review]
A couple comments/typo but otherwise LGTM!

**Comment by tomMoral** [comment]
Merging, thx @mathurinm !

**Pull request merged**